app/src/main/java/com/ataulm/artcollector/paintings/ui/PaintingsViewModel.kt

```diff
@@ -5,31 +5,32 @@ import android.arch.lifecycle.MutableLiveData
 import android.arch.lifecycle.ViewModel
 import com.ataulm.artcollector.paintings.domain.GetPaintingsUseCase
 import com.ataulm.artcollector.paintings.domain.Painting
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 internal class PaintingsViewModel @Inject constructor(
-        val paintings: LiveData<List<Painting>>
-) : ViewModel()
-
-internal class PaintingsLiveData @Inject constructor(
         private val getPaintings: GetPaintingsUseCase
-) : MutableLiveData<List<Painting>>() {
+) : ViewModel() {
+
+    private val _paintings = MutableLiveData<List<Painting>>()
+    val paintings: LiveData<List<Painting>> = _paintings
 
-    private var job: Job? = null
+    private val parentJob = Job()
+    private val coroutineScope = CoroutineScope(parentJob)
```

Comment:
I decided _not_ to provide a default dispatcher. This means that unless it's provided when using a coroutine builder, it'll use `Dispatchers.Default`.

Chris Banes' point about providing `Dispatchers.Main` as the default, since that'd be the default behaviour sans coroutines, is also legit, but IMO these should be _explicitly_ provided at the call site, and we shouldn't rely on a default.